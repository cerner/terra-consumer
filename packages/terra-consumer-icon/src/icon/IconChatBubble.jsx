/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M46 4H2v30h14l8 10 8-10h14z" ></path><path data-color="color-2" fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M14 14h20" ></path><path data-color="color-2" fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M14 24h12" ></path>
    </IconBase>
  );
};

SvgIcon.displayName = "IconChatBubble";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg"};

export default SvgIcon;
/* eslint-enable */
