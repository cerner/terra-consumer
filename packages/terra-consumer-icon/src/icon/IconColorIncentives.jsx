/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M11 12h4" ></path><circle data-color="color-2" fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" cx="13" cy="3" r="2" ></circle><path fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M10 21v2" ></path><path fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M16 21v2" ></path><path fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M16.5 8H9C7.75 5.042 5 6 5 6l.9 3.089A6.531 6.531 0 0 0 3.505 12H1v5h2.5a6.5 6.5 0 0 0 6 4h7a6.5 6.5 0 1 0 0-13z" ></path>
    </IconBase>
  );
};

SvgIcon.displayName = "IconColorIncentives";
SvgIcon.defaultProps = {"viewBox":"0 0 24 24","height":"24","width":"24","xmlns":"http://www.w3.org/2000/svg"};

export default SvgIcon;
/* eslint-enable */
