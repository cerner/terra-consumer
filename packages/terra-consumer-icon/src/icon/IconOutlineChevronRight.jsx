/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path fill="none" stroke="#444" strokeWidth="2" strokeLinecap="square" strokeMiterlimit="10" d="M14 4l20 20-20 20" ></path>
    </IconBase>
  );
};

SvgIcon.displayName = "IconOutlineChevronRight";
SvgIcon.defaultProps = {"className":"","viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg","isBidi":true};

export default SvgIcon;
/* eslint-enable */
