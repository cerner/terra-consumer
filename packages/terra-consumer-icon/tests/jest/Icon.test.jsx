import React from 'react';

// IconChatBubble by default, isBidi: false, isSpin: false
// IconColorIncentives by default, isBidi: false, isSpin: false
// IconOutlineChevronRight by default, isBidi: true, isSpin: false
import IconChatBubble from '../../src/icon/IconChatBubble';
import IconColorIncentives from '../../src/icon/IconColorIncentives';
import IconOutlineChevronRight from '../../src/icon/IconOutlineChevronRight';

describe('Icon', () => {
  describe('IconChatBubble', () => {
    it('should shallow IconBase', () => {
      const wrapper = shallow(<IconChatBubble />);
      expect(wrapper).toMatchSnapshot();
    });

    describe('height prop', () => {
      it('should have default height="1em"', () => {
        const wrapper = shallow(<IconChatBubble />);
        expect(wrapper.prop('height')).toEqual('1em');
      });
      it('should have height=50', () => {
        const wrapper = shallow(<IconChatBubble height="50" />);
        expect(wrapper.prop('height')).toEqual('50');
      });
      it('should have height="4em"', () => {
        const wrapper = shallow(<IconChatBubble height="4em" />);
        expect(wrapper.prop('height')).toEqual('4em');
      });
    });

    describe('width prop', () => {
      it('should have default width="1em"', () => {
        const wrapper = shallow(<IconChatBubble />);
        expect(wrapper.prop('width')).toEqual('1em');
      });
      it('should have width=50', () => {
        const wrapper = shallow(<IconChatBubble width="50" />);
        expect(wrapper.prop('width')).toEqual('50');
      });
      it('should have width="4em"', () => {
        const wrapper = shallow(<IconChatBubble width="4em" />);
        expect(wrapper.prop('width')).toEqual('4em');
      });
    });

    describe('focusable prop', () => {
      it('should have default focusable={false}', () => {
        const wrapper = shallow(<IconChatBubble />);
        expect(wrapper.prop('focusable')).toEqual(false);
      });
      it('should have focusable={true}', () => {
        const wrapper = shallow(<IconChatBubble focusable />);
        expect(wrapper.prop('focusable')).toEqual(true);
      });
    });

    describe('isBidi prop', () => {
      it('should have default isBidi={false}', () => {
        const wrapper = shallow(<IconChatBubble />);
        expect(wrapper.prop('isBidi')).toEqual(false);
      });
      it('should have isBidi={true}', () => {
        const wrapper = shallow(<IconChatBubble isBidi />);
        expect(wrapper.prop('isBidi')).toEqual(true);
      });
    });

    describe('isSpin prop', () => {
      it('should have default isSpin={false}', () => {
        const wrapper = shallow(<IconChatBubble />);
        expect(wrapper.prop('isSpin')).toEqual(false);
      });
      it('should have isSpin={true}', () => {
        const wrapper = shallow(<IconChatBubble isSpin />);
        expect(wrapper.prop('isSpin')).toEqual(true);
      });
    });

    describe('ariaLabel prop', () => {
      it('should have default ariaLabel={null}', () => {
        const wrapper = shallow(<IconChatBubble />);
        expect(wrapper.prop('ariaLabel')).toEqual(null);
      });
      it('should have ariaLabel={some value}', () => {
        const wrapper = shallow(<IconChatBubble ariaLabel="some value" />);
        expect(wrapper.prop('ariaLabel')).toEqual('some value');
      });
    });
  });

  describe('IconColorIncentives', () => {
    it('should shallow IconBase', () => {
      const wrapper = shallow(<IconColorIncentives />);
      expect(wrapper).toMatchSnapshot();
    });

    describe('height prop', () => {
      it('should have default height="1em"', () => {
        const wrapper = shallow(<IconColorIncentives />);
        expect(wrapper.prop('height')).toEqual('1em');
      });
      it('should have height=50', () => {
        const wrapper = shallow(<IconColorIncentives height="50" />);
        expect(wrapper.prop('height')).toEqual('50');
      });
      it('should have height="4em"', () => {
        const wrapper = shallow(<IconColorIncentives height="4em" />);
        expect(wrapper.prop('height')).toEqual('4em');
      });
    });

    describe('width prop', () => {
      it('should have default width="1em"', () => {
        const wrapper = shallow(<IconColorIncentives />);
        expect(wrapper.prop('width')).toEqual('1em');
      });
      it('should have width=50', () => {
        const wrapper = shallow(<IconColorIncentives width="50" />);
        expect(wrapper.prop('width')).toEqual('50');
      });
      it('should have width="4em"', () => {
        const wrapper = shallow(<IconColorIncentives width="4em" />);
        expect(wrapper.prop('width')).toEqual('4em');
      });
    });

    describe('focusable prop', () => {
      it('should have default focusable={false}', () => {
        const wrapper = shallow(<IconColorIncentives />);
        expect(wrapper.prop('focusable')).toEqual(false);
      });
      it('should have focusable={true}', () => {
        const wrapper = shallow(<IconColorIncentives focusable />);
        expect(wrapper.prop('focusable')).toEqual(true);
      });
    });

    describe('isBidi prop', () => {
      it('should have default isBidi={false}', () => {
        const wrapper = shallow(<IconColorIncentives />);
        expect(wrapper.prop('isBidi')).toEqual(false);
      });
      it('should have isBidi={true}', () => {
        const wrapper = shallow(<IconColorIncentives isBidi />);
        expect(wrapper.prop('isBidi')).toEqual(true);
      });
    });

    describe('isSpin prop', () => {
      it('should have default isSpin={false}', () => {
        const wrapper = shallow(<IconColorIncentives />);
        expect(wrapper.prop('isSpin')).toEqual(false);
      });
      it('should have isSpin={false} with override', () => {
        const wrapper = shallow(<IconColorIncentives isSpin={false} />);
        expect(wrapper.prop('isSpin')).toEqual(false);
      });
    });

    describe('ariaLabel prop', () => {
      it('should have default ariaLabel={null}', () => {
        const wrapper = shallow(<IconColorIncentives />);
        expect(wrapper.prop('ariaLabel')).toEqual(null);
      });
      it('should have ariaLabel={some value}', () => {
        const wrapper = shallow(<IconColorIncentives ariaLabel="some value" />);
        expect(wrapper.prop('ariaLabel')).toEqual('some value');
      });
    });
  });


  describe('IconOutlineChevronRight', () => {
    it('should shallow IconBase', () => {
      const wrapper = shallow(<IconOutlineChevronRight />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render IconBase with custom-class', () => {
      const wrapper = render(<IconOutlineChevronRight className="custom-class" />);
      expect(wrapper).toMatchSnapshot();
    });

    describe('height prop', () => {
      it('should have default height="1em"', () => {
        const wrapper = shallow(<IconOutlineChevronRight />);
        expect(wrapper.prop('height')).toEqual('1em');
      });
      it('should have height=50', () => {
        const wrapper = shallow(<IconOutlineChevronRight height="50" />);
        expect(wrapper.prop('height')).toEqual('50');
      });
      it('should have height="4em"', () => {
        const wrapper = shallow(<IconOutlineChevronRight height="4em" />);
        expect(wrapper.prop('height')).toEqual('4em');
      });
    });

    describe('width prop', () => {
      it('should have default width="1em"', () => {
        const wrapper = shallow(<IconOutlineChevronRight />);
        expect(wrapper.prop('width')).toEqual('1em');
      });
      it('should have width=50', () => {
        const wrapper = shallow(<IconOutlineChevronRight width="50" />);
        expect(wrapper.prop('width')).toEqual('50');
      });
      it('should have width="4em"', () => {
        const wrapper = shallow(<IconOutlineChevronRight width="4em" />);
        expect(wrapper.prop('width')).toEqual('4em');
      });
    });

    describe('focusable prop', () => {
      it('should have default focusable={false}', () => {
        const wrapper = shallow(<IconOutlineChevronRight />);
        expect(wrapper.prop('focusable')).toEqual(false);
      });
      it('should have focusable={true}', () => {
        const wrapper = shallow(<IconOutlineChevronRight focusable />);
        expect(wrapper.prop('focusable')).toEqual(true);
      });
    });

    describe('isBidi prop', () => {
      it('should have default isBidi={true}', () => {
        const wrapper = shallow(<IconOutlineChevronRight />);
        expect(wrapper.prop('isBidi')).toEqual(true);
      });
      it('should have isBidi={false} with override', () => {
        const wrapper = shallow(<IconOutlineChevronRight isBidi={false} />);
        expect(wrapper.prop('isBidi')).toEqual(false);
      });
    });

    describe('isSpin prop', () => {
      it('should have default isSpin={false}', () => {
        const wrapper = shallow(<IconOutlineChevronRight />);
        expect(wrapper.prop('isSpin')).toEqual(false);
      });
      it('should have isSpin={true}', () => {
        const wrapper = shallow(<IconOutlineChevronRight isSpin />);
        expect(wrapper.prop('isSpin')).toEqual(true);
      });
    });

    describe('ariaLabel prop', () => {
      it('should have default ariaLabel={null}', () => {
        const wrapper = shallow(<IconOutlineChevronRight />);
        expect(wrapper.prop('ariaLabel')).toEqual(null);
      });
      it('should have ariaLabel={some value}', () => {
        const wrapper = shallow(<IconOutlineChevronRight ariaLabel="some value" />);
        expect(wrapper.prop('ariaLabel')).toEqual('some value');
      });
    });
  });
});
