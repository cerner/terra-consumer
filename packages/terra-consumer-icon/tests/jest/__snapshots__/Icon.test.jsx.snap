// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon IconChatBubble should shallow IconBase 1`] = `
<IconBase
  ariaLabel={null}
  focusable={false}
  height="1em"
  isBidi={false}
  isSpin={false}
  viewBox="0 0 48 48"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M46 4H2v30h14l8 10 8-10h14z"
    fill="none"
    stroke="#444"
    strokeLinecap="square"
    strokeMiterlimit="10"
    strokeWidth="2"
  />
  <path
    d="M14 14h20"
    data-color="color-2"
    fill="none"
    stroke="#444"
    strokeLinecap="square"
    strokeMiterlimit="10"
    strokeWidth="2"
  />
  <path
    d="M14 24h12"
    data-color="color-2"
    fill="none"
    stroke="#444"
    strokeLinecap="square"
    strokeMiterlimit="10"
    strokeWidth="2"
  />
</IconBase>
`;

exports[`Icon IconColorIncentives should shallow IconBase 1`] = `
<IconBase
  ariaLabel={null}
  focusable={false}
  height="1em"
  isBidi={false}
  isSpin={false}
  viewBox="0 0 48 48"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M42 31.112a1 1 0 1 1 0-2c1.369 0 3-.52 3-3 0-1.637-1.018-1.973-1.222-2.025a1.008 1.008 0 0 1-.743-1.195.986.986 0 0 1 1.17-.758c.966.201 2.795 1.292 2.795 3.978 0 3.037-1.962 5-5 5z"
    fill="#D39B75"
  />
  <path
    d="M30 17H18.641c-2.36-4.749-6.848-4.682-8.97-3.944a1 1 0 0 0-.62 1.26l1.656 4.968A13.048 13.048 0 0 0 6.012 25H2a1 1 0 0 0-1 1v10a1 1 0 0 0 1 1h5.087A13.054 13.054 0 0 0 13 41.988V46a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-3h6v3a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-4.012C39.813 39.98 43 35.25 43 30c0-7.168-5.832-13-13-13z"
    fill="#EEBC99"
  />
  <path
    d="M30 23h-8a1 1 0 1 1 0-2h8a1 1 0 1 1 0 2z"
    fill="#D39B75"
  />
  <circle
    cx="26"
    cy="6"
    fill="#EFD358"
    r="5"
  />
  <path
    d="M14 28c-1.103 0-2-.897-2-2s.897-2 2-2 2 .897 2 2-.897 2-2 2zm0-1.999h.01H14zm0 0h.01H14zm0 0h.01H14zm0 0h.01H14zm0 0h.01H14zM14 26h.01H14zm0 0h.01H14zm0 0h.01H14z"
    fill="#444"
  />
</IconBase>
`;

exports[`Icon IconOutlineChevronRight should render IconBase with custom-class 1`] = `
<svg
  aria-hidden="true"
  class="icon is-bidi custom-class"
  focusable="false"
  height="1em"
  viewbox="0 0 48 48"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14 4l20 20-20 20"
    fill="none"
    stroke="#444"
    stroke-linecap="square"
    stroke-miterlimit="10"
    stroke-width="2"
  />
</svg>
`;

exports[`Icon IconOutlineChevronRight should shallow IconBase 1`] = `
<IconBase
  ariaLabel={null}
  className=""
  focusable={false}
  height="1em"
  isBidi={true}
  isSpin={false}
  viewBox="0 0 48 48"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14 4l20 20-20 20"
    fill="none"
    stroke="#444"
    strokeLinecap="square"
    strokeMiterlimit="10"
    strokeWidth="2"
  />
</IconBase>
`;
